### PR TITLE
Fix error with `ConcurrentHashMap` on Android <= 9

### LINKED ISSUE #1751: Java.Lang.Error: no non-static method "Ljava/util/concurrent/ConcurrentHashMap$KeyIterator;.hasNext()Z"
Sentry Issue: [ANDROID-SYMBOL-COLLECTOR-MOBILE-1Y](https://sentry.io/organizations/sentry/issues/3352915862/?referrer=github_integration)

```
Java.Lang.Error: no non-static method "Ljava/util/concurrent/ConcurrentHashMap$KeyIterator;.hasNext()Z"
  ?, in JniMethodInfo InstanceMethods.GetMethodID(JniObjectReference, string, string)
  ?, in bool IIteratorInvoker.get_HasNext()
...
(5 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
Fixes #1751

### CODE REVIEW

**Comment by mattjohnsonpint** [comment]
FYI, I will also report this upstream to the appropriate Microsoft repo. They might have a better workaround, or want to fix it under the hood.

File: src/Sentry/Android/Extensions/JavaExtensions.cs [hunk 1]
```diff
@@ -9,4 +12,19 @@ internal static class JavaExtensions
...
 
     public static Throwable ToThrowable(this Exception exception) => Throwable.FromException(exception);
+
+    public static IDictionary<TKey, TValue> WorkaroundKeyIteratorBug<TKey, TValue>(this IDictionary<TKey, TValue> dictionary)
+    {
+        // Workaround for https://github.com/getsentry/sentry-dotnet/issues/1751
+        // Java.Lang.Error: no non-static method "Ljava/util/concurrent/ConcurrentHashMap$KeyIterator;.hasNext()Z"
+        // Affects Android 9 ("Pie") and older (API 28)
+
+        if (AndroidBuild.VERSION.SdkInt > BuildVersionCodes.P)
```
Comment (bruno-garcia) [inline src/Sentry/Android/Extensions/JavaExtensions.cs:22]: @marandaneto welxome Matt to android development please 😅

**Review by bruno-garcia — APPROVED** [review]
Thanks Matt!!

**Comment by mattjohnsonpint** [comment]
Reported here: https://github.com/xamarin/java.interop/issues/1003

**Pull request merged**